src/signal.rs

```diff
@@ -1903,4 +1928,170 @@ mod menu_item {
         callback_guard!();
         f(&MenuItem::from_glib_none(this).downcast_unchecked());
     }
-}
\ No newline at end of file
+}
+
+pub trait AboutDialogSignals {
+    fn connect_activate_link<F>(&self, activate_func: F) -> u64
+        where F: Fn(&Self, &str) + 'static;
+}
+
+mod about_dialog {
+    use AboutDialog;
+    use std::mem::transmute;
+    use ffi::GtkAboutDialog;
+    use glib::object::Downcast;
+    use glib::signal::connect;
+    use glib::translate::*;
+    use IsA;
+    use libc::c_char;
+    use std::ffi::CStr;
+    use std::str;
+
+    impl super::AboutDialogSignals for AboutDialog {
+        fn connect_activate_link<F: Fn(&Self, &str) + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(&Self, &str) + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0, "activate-link",
+                    transmute(trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+    }
+
+    unsafe extern "C" fn trampoline<T>(this: *mut GtkAboutDialog, c_str: *const c_char,
+                                       f: &Box<Fn(&T, &str) + 'static>)
+    where T: IsA<AboutDialog> {
+        callback_guard!();
+        let buf = CStr::from_ptr(c_str).to_bytes();
+        let string = str::from_utf8(buf).unwrap();
+        f(&AboutDialog::from_glib_none(this).downcast_unchecked(), string);
+    }
+}
+
+pub trait CellAreaSignals {
+    fn connect_add_editable<F>(&self, add_editable_func: F) -> u64
+        where F: Fn(&Self, &CellRenderer, &CellEditable, &Rectangle, &str) + 'static;
+    fn connect_apply_attributes<F>(&self, apply_attributes_func: F) -> u64
+        where F: Fn(&Self, &TreeModel, &TreeIter, bool, bool) + 'static;
+    fn connect_focus_changed<F>(&self, focus_changed_func: F) -> u64
+        where F: Fn(&Self, &CellRenderer, &str) + 'static;
+    fn connect_remove_editable<F>(&self, remove_editable_func: F) -> u64
+        where F: Fn(&Self, &CellRenderer, &CellEditable) + 'static;
+}
+
+mod cell_area {
+    use CellArea;
+    use CellEditable;
+    use CellRenderer;
+    use TreeIter;
+    use TreeModel;
+    use Rectangle;
+    use std::mem::transmute;
+    use ffi::{GtkCellArea, GtkCellEditable, GtkCellRenderer, GtkTreeIter, GtkTreeModel};
+    use gdk_ffi::GdkRectangle;
+    use glib_ffi::gboolean;
+    use glib::object::Downcast;
+    use glib::signal::connect;
+    use glib::translate::*;
+    use IsA;
+    use libc::c_char;
+    use std::ffi::CStr;
+    use std::str;
+
+    impl super::CellAreaSignals for CellArea {
+        fn connect_add_editable<F>(&self, add_editable_func: F) -> u64
+        where F: Fn(&Self, &CellRenderer, &CellEditable, &Rectangle, &str) + 'static {
```

Comment:
Meh, I've missed the fact that the string parameter is a path. [Here's how we handle that](https://github.com/gtk-rs/gtk/blob/c40bc7abe7f922957d59c1df360d55f8e5687996/src/signal.rs#L1150-L1165). Applies to `focus-changed` as well.